src/tools/ceph_objectstore_tool.cc

```diff
@@ -646,23 +646,33 @@ int do_trim_pg_log(ObjectStore *store, const coll_t &coll,
 
   cerr << "Log bounds are: " << "(" << info.log_tail << ","
        << info.last_update << "]" << std::endl;
-
-  uint64_t max_entries = g_ceph_context->_conf->osd_max_pg_log_entries;
+  uint64_t max_entries = std::min<uint64_t>(g_ceph_context->_conf->osd_max_pg_log_entries,
+					    (info.last_update.version) ? (info.last_update.version - 1)
+					        : g_ceph_context->_conf->osd_max_pg_log_entries
+					    );
   if (info.last_update.version - info.log_tail.version <= max_entries) {
     cerr << "Log not larger than osd_max_pg_log_entries " << max_entries << std::endl;
     return 0;
```

Comment:
On a PG with 18M entries, we get
```
Log bounds are: (1314488'70894144,1314494'70896562]
Log not larger than osd_max_pg_log_entries 10000
Finished trimming pg log
```

Can we also check len(dups) before aborting the trim  ?

Reply:
@NitzanMordhai ^